waspc/cli/src/Wasp/Cli/Command/CreateNewProject.hs

```diff
@@ -103,5 +127,71 @@ writeMainWaspFile waspProjectDir (ProjectInfo projectName appName) = IOUtil.writ
+createProjectFromTemplate absWaspProjectDir ProjectInfo {_appName = appName, _projectName = projectName, _template = maybeTemplateName} = do
+  cliSendMessageC $ Msg.Start "Creating project from template..."
+
+  dirWhereProjectIsCreated <- getDirWhereProjectIsCreated
+  templatePath <- getPathToRemoteTemplate maybeTemplateName
+
+  let projectDir = projectName
+
+  fetchTemplate templatePath projectDir dirWhereProjectIsCreated
+  ensureTemplateWasFetched
+
+  replacePlaceholdersInWaspFile
+  where
+    getDirWhereProjectIsCreated :: Command (Path System Abs (Dir dir))
+    getDirWhereProjectIsCreated = do
+      (liftIO getCurrentDirectory <&> parseAbsDir) >>= \case
+        Left err -> throwProjectCreationError $ "Failed to parse absolute path to current working directory: " ++ show err
+        Right result -> return result
+
+    getPathToRemoteTemplate :: Maybe String -> Command String
+    getPathToRemoteTemplate = \case
+      Just templateName -> return $ waspTemplatesRepo ++ "/" ++ templateName
+      Nothing -> throwProjectCreationError "Template name is not provided."
+      where
+        -- gh: prefix means Github repo
+        waspTemplatesRepo = "gh:wasp-lang/starters"
+
+    fetchTemplate :: String -> String -> Path System Abs (Dir dir) -> Command ()
+    fetchTemplate templatePath projectDir dirWhereProjectIsCreated = do
+      liftIO executeCmd >>= \case
+        ExitSuccess -> return ()
+        ExitFailure _ -> throwProjectCreationError "Failed to create project from template."
+      where
+        executeCmd = newChan >>= runNodeCommandAsJob dirWhereProjectIsCreated "npx" args J.Cli
+        args = ["giget@latest", templatePath, projectDir]
+
+    -- gitget doesn't fail if the template dir doesn't exist in the repo, so we need to check if the directory exists.
+    ensureTemplateWasFetched :: Command ()
+    ensureTemplateWasFetched = do
+      whenM
+        (liftIO $ IOUtil.isDirectoryEmpty absWaspProjectDir)
+        $ throwProjectCreationError "Are you sure that the template exists? 🤔 Check the list of templates here: https://github.com/wasp-lang/starters"
+
+    replacePlaceholdersInWaspFile :: Command ()
+    replacePlaceholdersInWaspFile = liftIO $ do
+      mainWaspFileContent <- IOUtil.readFileStrict absMainWaspFile
+
+      let replacedContent =
+            foldl
+              (\acc (placeholder, value) -> T.replace (T.pack placeholder) (T.pack value) acc)
+              mainWaspFileContent
+              placeholders
+
+      IOUtil.writeFileFromText absMainWaspFile replacedContent
+      where
+        absMainWaspFile = absWaspProjectDir </> [relfile|main.wasp|]
+        placeholders =
+          [ ("__waspAppName__", appName),
+            ("__waspProjectName__", projectName),
+            ("__waspVersion__", waspVersionStr)
+          ]
+
+waspVersionStr :: String
+waspVersionStr = "^%s" `printf` show WV.waspVersion
```

Comment:
I think in theory this is not version, but something like version range? In our SemanticVersion.hs module this stuff is relatively clear so maybe we can understand it from there. So if we can make it better let's do it, if not then ok.

Reply:
It's defined like this:
```hs
waspVersion :: SV.Version
waspVersion = SV.Version (toEnum major) (toEnum minor) (toEnum patch)
  where
    DV.Version [major, minor, patch] _ = Paths_waspc.version
```

I'll call it `waspCompilerVersionStr`

Reply:
But we are adding `^` in front ... that's true